fission/spec.go

```diff
@@ -388,6 +388,19 @@ func (fr *FissionResources) validate() error {
 	// we do not error on unreferenced functions (you can call a function through workflows,
 	// `fission function test`, etc.)
 
+	// Index envs, warn on functions referencing an environment for which spes does not exist
+	environments := make(map[string]bool)
```

Comment:
There is a go convention which says that if you are using a map just for the keys. You should use `map[string]struct{}` since that makes it explicit that the value is not significant.